### PR TITLE
Fix some build issues in gcc and clang on 18.04

### PR DESCRIPTION
**Pull request by yakman2020**
Fixes some build breaks on ubuntu bionic (18.04).

### CODE REVIEW

**Comment by yakman2020** [comment]
bors try

File: 3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,12 @@
 # Licensed under the MIT License.
 
 # Note: order does matter
+if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  # Enables all the warnings about constructions that some users consider questionable,
```
Comment (andschwa) [inline 3rdparty/CMakeLists.txt]: I am not sure I understand this comment.

This should be saying: "While we enable `-Werror` globally in our CMake build, we do not want to fix all the warnings in third-party code. Thus we disable it here by adding `-Wno-error` for all targets under this folder."

Comment (andschwa) [reply 3rdparty/CMakeLists.txt]: Also, this change appears separate from the change below. If we're going to disable treating warnings as errors for third-party code, shouldn't we then also remove all the code disabling specific warnings in third-party code?

Comment (anakrish) [reply 3rdparty/CMakeLists.txt]: @yakman2020 @andschwa We do have our own code in 3rdparty (e.g. we customize libunwind that way; in future we would need to tweak libcxxrt as well to interface with the debugger when throwing exceptions). We want out code to be warnings free. So do you think -Wno-error in 3rdparty top-level is the right way?

Comment (yakman2020) [reply 3rdparty/CMakeLists.txt]: I think if we are heavily modifying code as you say, we should not have it in 3rdparty.

**Comment by yakman2020** [comment]
bors try

**Comment by yakman2020** [comment]
bors try

**Comment by yakman2020** [comment]
bors try

**Comment by yakman2020** [comment]
bors try

**Pull request merged**